assignments/closure.js

```diff
@@ -1,20 +1,55 @@
 // ==== Challenge 1: Write your own closure ====
 // Write a simple closure of your own creation.  Keep it simple!
+function outer(data) {
+  return function inner() {
+    console.log(data);
+    return data;
+  }
+}
 
+const inner = outer('hello');
+inner();
 
 // ==== Challenge 2: Create a counter function ====
 const counter = () => {
   // Return a function that when invoked increments and returns a counter variable.
+  let count = 0;
+  return () => {
+    // ==> Not using ++ because: Per the eslint documentation, unary increment and decrement statements are subject to automatic semicolon insertion and can cause silent errors with incrementing or decrementing values within an application.
```

Comment:
Really nice spotting this, sometimes `++` can cause troubles. This was one of the reasons why it was completely removed from Swift programming language for example.